optimization/optimization.py

```diff
@@ -0,0 +1,351 @@
+
+                filtered_indexes = indexes[:, filtered]
+
+                filter_total = np.append(filter_total, np.array(filtered_indexes), 1)
+
+                filtered_indexes = None
+                indexes = None
+                filtered = None
+        
+        # Generate all the indexes all at once
+        else:
+            indexes = generate_indexes(tactile_based_selector.indexes_list)
+            indexes = jnp.array(indexes)
+
+            filtered = eval_triplets(indexes, points, landmarks, tactile_based_selector.number_of_sensors, comb_list, distance_threshold)
+
+            filtered_indexes = indexes[:, filtered]
+            filter_total = np.append(filter_total, np.array(filtered_indexes), 1)
+            filtered_indexes = None
+            indexes = None
+            filtered = None
+
+    indexes = filter_total
+    points = jnp.array(points)
+    landmarks = jnp.array(landmarks)
+
+    boolean_loop = False
+    boolean_triplets = True
+
+    if tactile_based_selector.number_of_sensors != 0:
+
+        # Check whether the number of triplets to be studied is appropriate
+        while boolean_triplets:
+
+            filter = eval_triplets(indexes, points, landmarks, tactile_based_selector.number_of_sensors, comb_list, distance_threshold)
+
+
+            filtered_indexes = indexes[:, filter]
+            print(filtered_indexes.shape)
+
+            if filtered_indexes.shape[1] < 5000 and filtered_indexes.shape[1] >0:
+                boolean_triplets = False
+
+            elif filtered_indexes.shape[1] > 5000 and boolean_loop:
+                boolean_triplets = False
+
+            elif filtered_indexes.shape[1] == 0:
+                distance_threshold += 0.0001
+                boolean_loop = True
+                filter = None
+            else:
+                distance_threshold -= 0.0005
+                filtered_indexes = None
+    else:
+        filtered_indexes = indexes
+
+    filtered_indexes = jnp.array(filtered_indexes)
+    print("Remaining indexes shape: ")
+    print(filtered_indexes.shape)
+
+    # Loss
+    def eval_item(item, points, landmarks, twist):
+        T = SO3.exp(twist[3:6]).as_matrix()
+        error = 0
+        for i in range(landmarks.shape[1]):
+            error += jnp.linalg.norm((T @ points[:, item[i]] + twist[0:3]) - landmarks[:, i])
+
+        return error / landmarks.shape[1]
+
+    def eval_items(twist, filtered_index):
+        return eval_item(filtered_index, points, landmarks, twist)
+
+    def eval_items_final(twists):
+        return vmap(eval_items, in_axes=(1,1))(twists, filtered_indexes)
+
+    g = grad(eval_items)
+
+    def grads(twists):
+        return vmap(g, in_axes=(1,1))(twists, filtered_indexes)
+
+    grads = jit(grads)
+
+    # Define the iteration of the optimization process.
+    # 700 represent an heuristic solution
+    iters = 700
```

Comment:
Can we move this to the configuration file?
```suggestion
    iters = 700
```